manual/trees/index.md

```diff
@@ -7,178 +7,145 @@ toc: true
+### The tree and its data
 
-- [TBranch](https://root.cern/doc/master/classTBranch.html){:target="_blank"}: Organizes columns, i.e. branches, of a `TTree`.
+A `TTree` behaves like an array of a data structure that resides on storage - except for one entry (or row, in database language).
+That entry is accessible in memory: you can load any tree entry, ideally sequentially.
+You can provide your own storage for the values of the columns of the current entry, in the form of variables.
+In this case you have to tell the `TTree` about the addresses of these variables; either by calling [`TTree::SetBranchAddress()`](https://root.cern/doc/master/classTTree.html#a39b867210e4a77ef44917fd5e7898a1d), or by passing the variable when creating the branch for writing.
+When "filling" (writing) the `TTree`, it will read the values out of these variables;
+when reading back a `TTree` entry, it will write the values it read from storage into your variables.
 
-- [TChain](https://root.cern/doc/master/classTChain.html){:target="_blank"}: A list of ROOT files containing `TTree` objects.
+### Branches and leaves
 
+A tree consists of a list of independent columns, called branches. A branch can contain values of any fundamental type, C++ objects known to ROOT's type system, or collections of those.
+Branches are represented by {% include ref class="TBranch" %} and its derived classes.
 
-## Working with trees
+While `TBranch` represent structure, objects inheriting from {% include ref class="TLeaf" %} contain the actual data.
+Originally, any columnar data was contained in a `TLeaf`; these days, some of the `TBranch`-derived classes contain data themselves, such as {% include ref class="TBranchElement" %}.
 
-ROOT offers many possibilities to work with trees, for example:
+### `TNtuple`, the high-performance spread-sheet
 
-- [Creating a tree](#creating-a-tree)
-- [Creating a tree from a folder structure](#creating-a-tree-from-a-folder-structure)
-- [Filling a tree](#filling-a-tree)
-- [Writing a tree](#writing-a-tree)
-- [Printing the summary of a tree](#printing-the-summary-of-a-tree)
-- [Showing an entry of a tree](#showing-an-entry-of-a-tree)
-- [Scanning trees](#scanning-trees)
+For convenience, ROOT also provides the {% include ref class="TNtuple" %} class which is a {% include ref class="TTree" %} that is limited to contain floating-point numbers only.
 
-### Creating a tree
+## Writing a tree
 
-- Use the {% include ref class="TTree" %} constructor to create a tree.
+When writing a `TTree` you first want to create a `TFile`
+(see → [ROOT files]({{ '/manual/root_files' | relative_url }}).
+Then construct the `TTree` to be stored in the file; we will later add branches to the tree.
 
 _**Example**_
 
 {% highlight C++ %}
-   TTree t("MyTree","Example Tree");
+std::unique_ptr<TFile> myFile( TFile::Open("file.root", "RECREATE") );
+auto tree = std::make_unique<TTree>("tree", "The Tree Title");
```

Comment:
The "unique" part is misleading since there is shared owership between the user and the `TFile`.  A raw pointer is likely adequate here as *usually* the user should not have to worry about deleting the TTree (but they can if they want to).

Reply:
I think the key is:

> but they can if they want to

That means we should use the best practices pattern, and that's a stack variable - which is awkward because of ROOT's memory management requirements, and thus the secon best option is a `unique_ptr`. Do you see a *problem* with that, or is it "just" misleading?

Reply:
it is just misleading.